lib/src/over_react_redux/devtools/middleware.dart

```diff
@@ -0,0 +1,143 @@
+@JS()
+library over_react_redux.devtools.middleware;
+
+import 'dart:convert';
+import 'dart:js_util';
+
+import 'package:react/react_client/js_backed_map.dart';
+import 'package:redux/redux.dart';
+import 'package:redux_dev_tools/redux_dev_tools.dart';
+import 'package:js/js.dart';
+
+@JS()
+@anonymous
+class ReduxDevToolsExtensionConnection {
+  external factory ReduxDevToolsExtensionConnection();
+  external subscribe(Function listener);
+  external send([action, state]);
+  external init([initalState]);
+}
+
+@JS('__REDUX_DEVTOOLS_EXTENSION__.connect')
+external ReduxDevToolsExtensionConnection reduxExtConnect([dynamic options]);
+
+class OverReactReduxDevToolsMiddleware extends MiddlewareClass {
+  bool devToolsExtensionFound = false;
+  Store _store;
+  ReduxDevToolsExtensionConnection devToolsExt;
+  OverReactReduxDevToolsMiddleware() {
+    try {
+      devToolsExt = reduxExtConnect();
+      devToolsExtensionFound = true;
+    } catch(e) {
+      print('Unable to connect to the redux dev tools browser extension. \nPlease install it... \nChrome: https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en \nFirefox: https://addons.mozilla.org/en-US/firefox/addon/reduxdevtools/');
+    }
+
+  }
+
+  set store(Store v) {
+    _store = v;
+    devToolsExt.init(_encodeForTransit(store.state));
+    devToolsExt.subscribe(handleEventFromRemote);
+  }
+
+  get store => _store;
+
+  dynamic _encodeForTransit(dynamic content) {
+    try {
+    return jsify(jsonDecode(jsonEncode(content)));
+    } catch (e) {
+      print('You must implement a `toJson` method in your state and actions in order to view state changes in the redux dev tools.');
+    }
+  }
+
+  dynamic _encodeActionForTransit(dynamic action) {
+    try {
+      return _encodeForTransit({'type': _getActionType(action), 'payload': action});
+    } on Error {
```

Comment:
Honestly, I don't know. lol it was copy pasta from the other middlware

Reply:
That logic definitely makes sense. Let me know if I still missed something, but the redundancy of the `try`/`catch` (or `on`) between `_encodeForTransit` and `_encodeActionForTransit` still seems potentially problematic. I made this Dartpad example to kinda show what I mean: https://dartpad.dartlang.org/3db9928f1215feef94853a7cc169e00a

Because `_encodeForTransit` has a try/catch, won't any error at that level be swallowed, making the `on Error` in `_encodeActionForTransit` not really do anything?

Reply:
Good catch damn, yeah the inner try/catch doesnt re-throw so it will never error...

https://dartpad.dartlang.org/f8b7f141d10e033db487e971fdd0a852

thoughts on this approach?

Reply:
➕ 1️⃣